Source/gs/GSH_Vulkan/GSH_VulkanMemoryUtils.cpp

```diff
@@ -80,9 +80,9 @@ void CMemoryUtils::Memory_Write16(Nuanceur::CShaderBuilder& b, Nuanceur::CArrayU
 	AtomicOr(memoryBuffer, wordAddress, valueWord);
 }
 
-void CMemoryUtils::Memory_Write16(Nuanceur::CShaderBuilder& b, Nuanceur::CArrayUcharValue memoryBuffer, Nuanceur::CIntValue address, Nuanceur::CUintValue value)
+void CMemoryUtils::Memory_Write16(Nuanceur::CShaderBuilder& b, Nuanceur::CArrayUshortValue memoryBuffer, Nuanceur::CIntValue address, Nuanceur::CUintValue value)
 {
-	Store(memoryBuffer, address, ToUshort(value));
+	Store(memoryBuffer, address / NewInt(b, 2), ToUshort(value));
```

Comment:
This needs some tweaking, though its still notwhats causing `GSH_VulkanDraw` issue